### PR TITLE
doc: doc and version updates for 2.5.99-dev1

### PR DESCRIPTION
**Pull request by peknis**
Doc and version updates for 2.5.99-dev1 release

### CODE REVIEW

File: VERSION [hunk 1]
```diff
@@ -1 +1 @@
-2.5.99
+2.5.99-dev1
```
Comment (divipillai) [inline VERSION:1]: I think we don't update the version file: For example, https://github.com/nrfconnect/sdk-nrf/pull/12078/files.

**Review by cvinayak — APPROVED** [review]

File: VERSION (hunk 1 above)
Comment (divipillai) [reply VERSION:1]: Discussed offline. please disregard this comment.

File: doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst [hunk 2]
```diff
@@ -102,13 +55,13 @@ The following list summarizes the most important changes inherited from the upst
    * New device types:
 
      * Refidgerator
```
Comment (umapraseeda) [inline doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst:57]:
```suggestion
     * Refrigerator
```

File: doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst [hunk 3]
```diff
@@ -588,16 +335,12 @@ Trusted Firmware-M
...
-=====
-
-|no_changes_yet_note|
-
 Documentation
 =============
 
 * Updated:
 
+  * Replaced the Welcome to the |NCS| page with the existing :ref:`ncs_introduction` page.
```
Comment (divipillai) [inline doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst:584]: May be:
```suggestion
  * Removed the Welcome to the |NCS| page.
    This page is replaced with existing :ref:`ncs_introduction` page.
```
As a new bullet point. This is to match with release notes structure we have.

**Review by divipillai — APPROVED** [review]

File: doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst [hunk 4]
```diff
@@ -604,3 +587,6 @@ Documentation
   * The :ref:`ug_nrf9160_gs` by replacing the Updating the DK firmware section with a new :ref:`nrf9160_gs_installing_software` section.
     This new section includes steps for using Quick Start, a new application in `nRF Connect for Desktop`_ that streamlines the getting started process with the nRF91 Series DKs.
   * Updated the :ref:`tfm_enable_share_uart` section on :ref:`ug_nrf9160`.
```
Comment (umapraseeda) [inline doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst:589]: Also, please remove the extra colon after section in the first entry in this section.
```suggestion
  * The :ref:`tfm_enable_share_uart` section on :ref:`ug_nrf9160`.
```

**Review by umapraseeda — APPROVED** [review]

**Review by LuDuda — APPROVED** [review]

**Pull request merged**